File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        int n = matrix.size(), m = matrix[0].size();

        int c0 = 1;

    // Traverse the arr and mark first 
    // cell of each row and column
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                if (matrix[i][j] == 0) {

                // mark i-th row
                   matrix[i][0] = 0;

                // mark j-th column
                    if (j == 0)
                        c0 = 0;
                    else
                        matrix[0][j] = 0;
                }
            }
        }

    // Traverse and mark the matrix from 
    // (1, 1) to (n - 1, m - 1)
        for (int i = 1; i < n; i++) {
             for (int j = 1; j < m; j++) {

            // Check for col & row
                if (matrix[i][0] == 0 || matrix[0][j] == 0) {
                    matrix[i][j] = 0;
                }
            }
        }

    // Mark the first row
        if (matrix[0][0] == 0) {
            for (int j = 0; j < m; j++)
                matrix[0][j] = 0;
        }
  
    // Mark the first column
        if (c0 == 0) {
            for (int i = 0; i < n; i++)
                matrix[i][0] = 0;
        }
    }
};

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.py
class Solution:
    def setZeroes(self, mat: List[List[int]]) -> None:
        n=len(mat)
        m=len(mat[0])
        row=[0]*n
        col=[0]*m
        for i in range(n):
            for j in range(m):
                if mat[i][j]==0:
                    row[i]=1
                    col[j]=1
        for i in range(n):
            for j in range(m):
                if row[i]==1 or col[j]==1:
                    mat[i][j]=0